core/common/src/main/java/alluxio/resource/DynamicResourcePool.java

```diff
@@ -275,6 +276,15 @@ public DynamicResourcePool(Options options) {
         }
       }
     }, options.getInitialDelayMs(), options.getGcIntervalMs(), TimeUnit.MILLISECONDS);
+
+    registerGauges();
+  }
+
+  private void registerGauges() {
+    MetricsSystem.registerGaugeIfAbsent(
+        MetricsSystem.getMetricName("ResourcePoolSize_" + getClass().getName() + "_"
```

Comment:
shall we make this into a util, so other resource pool can follow the same convention to create metrics?